sherpa/astro/ui/tests/test_astro_ui.py

```diff
@@ -714,6 +714,23 @@ def read_func(filename):
     save_arrays(colnames=colnames, fits=False, read_func=read_func)
 
 
+
+@requires_fits
+@pytest.mark.parametrize("ascii_type", [False, True])
+def test_save_arrays_colmismatch_errs(ascii_type):
```

Comment:
You can use the tmp_path fixture to create a file name - so something like

```python
def test_save_arrays_colmismatch_errs(ascii_type, tmp_path):
    outfile = tmp_path / 'bogus.dat'
    outfile = str(outfile)  # Sherpa I/O doesn't handle Path objects
```

but I guess the point is that we actually don't expect the file to get created so is it worth the effort?